### PR TITLE
Margin ft

### PR DESCRIPTION
**Pull request by kilu01**
-  CRUD margin order
-  merge from huobi-futures-api to unify structure as a whole

### CODE REVIEW

File: lib/ex_huobi/margin/rest/order.ex [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  # ExHuobi.Margin.Rest.Order.cancel_order(70469904946)
+  def cancel_order(order_id, config \\ nil) do
+    case HTTPClient.post(
+           @margin_endpoint,
+           "/v1/order/orders/#{order_id}/submitcancel",
+           "",
+           config
+         ) do
+      {:ok, data} ->
```
Comment (sirius248) [inline lib/ex_huobi/margin/rest/order.ex:51]: what the data here mean? is this order_id ?

File: lib/ex_huobi/margin/rest/order.ex [hunk 2]
```diff
@@ -0,0 +1,80 @@
+defmodule ExHuobi.Margin.Rest.Order do
```
Comment (sirius248) [inline lib/ex_huobi/margin/rest/order.ex:1]: Can you add the unit test for this module?

File: lib/ex_huobi/margin/rest/account.ex [hunk 3]
```diff
@@ -0,0 +1,37 @@
+defmodule ExHuobi.Margin.Rest.Account do
```
Comment (sirius248) [inline lib/ex_huobi/margin/rest/account.ex:1]: Can you add unit tests for this module also ?

File: lib/ex_huobi/futures/private.ex [hunk 4]
```diff
@@ -0,0 +1,26 @@
+defmodule ExHuobi.Futures.Private do
```
Comment (sirius248) [inline lib/ex_huobi/futures/private.ex:1]: Unit tests for this also

Comment (sirius248) [reply lib/ex_huobi/futures/private.ex:1]: Can you add the typespec for all of the public functions https://hexdocs.pm/elixir/typespecs.html

File: lib/ex_huobi/margin/rest/account.ex (hunk 3 above)
Comment (sirius248) [reply lib/ex_huobi/margin/rest/account.ex:1]: Can you add the typespec for all of the public functions https://hexdocs.pm/elixir/typespecs.html

File: lib/ex_huobi/rest/http_client.ex [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      {:ok, %{"status" => "ok", "data" => data}} ->
+        {:ok, data}
+
+      {:error, error} ->
+        {:error, {:poison_decode_error, error}}
+    end
+  end
+end
+
+# defmodule ExHuobi.Http.Client do
```
Comment (sirius248) [inline lib/ex_huobi/rest/http_client.ex]: If this it not use, can you remove it.

**Comment by kilu01** [comment]
close this one due to change code structure

**Pull request closed without merging**